frontend/services/entities/hosts.ts

```diff
@@ -218,7 +218,7 @@ export default {
     const { HOSTS_DELETE } = endpoints;
     return sendRequest("POST", HOSTS_DELETE, {
       filters: {
-        query,
+        query: query || undefined, // Critical: Prevents empty string passed to API
```

Comment:
```suggestion
        query: query || undefined, // Do not send `query: ""` since it will be interpreted as no-filter. (Meaning select all hosts). Either have a filter or send nothing.
```

Reply:
@RachelElysia just to be clear:
Sending `"filter": {}` will mean ALL HOSTS.
Will this code above not send it like this?

Reply:
Maybe it's better to add a deliberate ` query: "NO_QUERY"` ?

Reply:
CC: @mostlikelee

Reply:
This may be confusing for customers using the API.  When we add support for more filters, "all hosts" may look something like this:

```
  "filters": {
    "query": "NO_QUERY",
    "status": "NO_STATUS",
    "os_name": "NO_NAME",
     +30 filters....
   }
```

I think we want to be as clear as possible when doing something as destructive as "delete all hosts"

Reply:
<<Thread resolved in our meeting>>